server/app/services/email/graph/GraphApiEmailClient.java

```diff
@@ -0,0 +1,128 @@
+package services.email.graph;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.microsoft.graph.models.BodyType;
+import com.microsoft.graph.models.EmailAddress;
+import com.microsoft.graph.models.ItemBody;
+import com.microsoft.graph.models.Message;
+import com.microsoft.graph.models.Recipient;
+import com.microsoft.graph.users.item.sendmail.SendMailPostRequestBody;
+import com.microsoft.kiota.ApiException;
+import com.typesafe.config.Config;
+import io.prometheus.client.Histogram;
+import java.util.ArrayList;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.Environment;
+import services.cloud.azure.Credentials;
+import services.email.EmailSendClient;
+import services.monitoring.EmailSendMetrics;
+import software.amazon.awssdk.http.HttpStatusCode;
+
+/** GraphApiEmailClient provides methods to send email notifications through Microsoft Graph API. */
+@Singleton
+public class GraphApiEmailClient implements EmailSendClient {
+  public static final String AZURE_SENDER_CONF_PATH = "email.sender";
+  private static final Logger logger = LoggerFactory.getLogger(GraphApiEmailClient.class);
+
+  private final EmailSendMetrics emailSendMetrics;
+  private final GraphApiClientInterface client;
+  private final Environment environment;
+  private final String sender;
+
+  @Inject
+  public GraphApiEmailClient(
+      Credentials credentials,
+      Config config,
+      Environment environment,
+      EmailSendMetrics emailSendMetrics) {
+    this.environment = checkNotNull(environment);
+    this.sender = checkNotNull(config).getString(AZURE_SENDER_CONF_PATH);
+    this.emailSendMetrics = emailSendMetrics;
+
+    if (environment.isProd()) {
+      client = new GraphApiClient(credentials);
+    } else {
+      client = new TestGraphApiClient();
+    }
+  }
+
+  @Override
+  public void send(String toAddress, String subject, String bodyText) {
+    send(ImmutableList.of(toAddress), subject, bodyText);
+  }
+
+  @Override
+  public void send(ImmutableList<String> toAddresses, String subject, String bodyText) {
+    if (toAddresses.isEmpty()) {
```

Comment:
Yeah it is - https://github.com/civiform/civiform/blob/e756bec012112d6bcea9c70b154a919f40ef0084/server/app/services/email/aws/SimpleEmail.java#L77. I think there may be cases where an email isn't set because the applicant is a guest, etc. but it may be helpful to at least know how often this is happening by adding some logging or a counter metric for this. Thoughts?